### ISSUE TITLE
Controls' texts catch selection (focus?) while dragging on Safari iOS 13.1.3

### ISSUE THREAD

**Issue by przelu** [Leaflet/Leaflet#6872]
- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
- open leafletjs.com on mobile
- put your finger on the map and wait for a moment
Note: holding finger on the is *not* dragging per se, yet this also happens to me while dragging *slowly*; users may also want to pause while dragging, as they explore the map. 

**Expected behavior**
the '+' in the zoom control or text in attribution control are not selected

**Current behavior**
see screenshot
![image](https://user-images.githubusercontent.com/33295482/67663457-ddd13800-f965-11e9-9fb4-06c09d988cab.png)

**Environment**
- Leaflet version: 1.5.1
- Browser (with version): Safari mobile
- OS/Platform (with version): iOS 13.1.3

**Additional context**
I got a React app, using react-leaflet.js.org, but I can also reproduce this on leafletjs.com, as mentioned above 

- [x] this example is as simple as possible
- [x] this example does not rely on any third party code

**Comment by Malvoz** [comment]
The controls are also highlighted on desktop when double-clicking the map area with the mouse pointer.

<a href="https://geogratis.gc.ca/mapml/en/osmtile/osm/"><img src="https://user-images.githubusercontent.com/26493779/76650049-1d3da100-6562-11ea-9f0c-fdb87b001f73.png" alt="Controls incorrectly selected"></a>

One potential solution could be to disallow text selection using CSS [`user-select`](https://developer.mozilla.org/en-US/docs/Web/CSS/user-select), e.g.:

```CSS
.leaflet-control a {
    -webkit-user-select: none;
       -moz-user-select: none;
        -ms-user-select: none;
            user-select: none;
}
```

**Comment by johnd0e** [comment]
@Malvoz
It seems not good to disable user-selection unconditionally.
That is right for controls like zoom, but wrong for others, like attribution.

Even worse: with proposed code when user tries to select whole text - only some fragments get selected:
![leaflet-doubleclick](https://user-images.githubusercontent.com/26493779/77778108-d0bd8f80-7050-11ea-80f4-98acd9cdf36f.png)

**Comment by Malvoz** [comment]
I agree. My "potential solution" derived from rather quickly looking into how other map tools have handled this. Google Maps (embed and API) seem content in disabling selection for attribution altogether, Bing Maps is using `pointer-events: none` (with `auto` for links) which seems less reliable.

And even _if_ you'd be okay with disabling text selection for controls and attribution, there would still be a problem: content outside of the map viewer (unless it's embedded in an iframe) would be the next available target for text selection.

So I don't think there's a quick CSS solution here, if only we had [`user-select: contain`](https://drafts.csswg.org/css-ui-4/#valdef-user-select-contain) ([caniuse](https://caniuse.com/#feat=mdn-css_properties_user-select_contain)), for e.g. `.leaflet-map-pane`).

**Comment by johnd0e** [comment]
I see similar behavior on taphold (contextmenu) in android Chrome: 
https://pomf.pyonpyon.moe/jvlscs.html

**Comment by johnd0e** [comment]
To do: check if #7059 can help.

**Comment by johnd0e** [comment]
As appeared, `tap` handler **was** able to prevent unwanted selection, with this code:
https://github.com/Leaflet/Leaflet/blob/32ea41baaa93adcefd34006a12ebe3b4ee7fda7d/src/map/handler/Map.Tap.js#L40-L43

This is not working atm, because touch events handling is currently emulated using pointer events (which was not intended btw, and still could be changed, see #7077).

Anyway, I do not feel that preventing `touchstart` is right way.


@Malvoz
> if only we had [`user-select: contain`](https://drafts.csswg.org/css-ui-4/#valdef-user-select-contain) ([caniuse](https://caniuse.com/#feat=mdn-css_properties_user-select_contain)),

But we could try to emulate that behavior, adding some class on `touchstart`, and removing it on `touchend/cancel`..

**Comment by Malvoz** [comment]
> > if only we had [`user-select: contain`](https://drafts.csswg.org/css-ui-4/#valdef-user-select-contain) ([caniuse](https://caniuse.com/#feat=mdn-css_properties_user-select_contain)),
>
> But we could try to emulate that behavior

Github has a polyfill: https://github.com/github/user-select-contain-polyfill. Perhaps Leaflet has a policy not to use polyfills, in which case it may serve as inspiration.

**Comment by Malvoz** [comment]
https://github.com/Leaflet/Leaflet/issues/5449#issuecomment-1133876106:

> [SVG icons](https://github.com/Leaflet/Leaflet/discussions/7729#discussioncomment-2599910) wouldn't catch text selection.
